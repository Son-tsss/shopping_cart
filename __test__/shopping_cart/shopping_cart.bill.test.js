import ShoppingCart from '../../src/shopping_cart/shopping_cart_actions'

it('should return correct bill', () => {
    const shoppingCart = new ShoppingCart

    const items = [
        {
            id: "i1",
            price: 100
        }, {
            id: "i2",
            price: 200
        }
    ]

    shoppingCart.addItem(items[0])
    shoppingCart.addItem(items[1])

    shoppingCart.applyDiscountToItem(0.5, items[0].id)
    shoppingCart.incrementItem(items[1].id)

    shoppingCart.applyDiscountToCart(0.75)

    const real = shoppingCart.getBill()

    expect(real).toMatchSnapshot()
})
