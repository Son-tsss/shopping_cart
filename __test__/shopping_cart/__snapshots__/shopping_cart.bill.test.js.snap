// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return correct bill 1`] = `
Object {
  "discount": 0.75,
  "items": Array [
    Object {
      "count": 1,
      "discount": 0.5,
      "finalPrice": 50,
      "id": "i1",
      "price": 100,
    },
    Object {
      "count": 2,
      "discount": 0.75,
      "finalPrice": 300,
      "id": "i2",
      "price": 200,
    },
  ],
  "price": 500,
  "total": 350,
}
`;
